scripts/mainsite.js

```diff
@@ -0,0 +1,213 @@
+var lastStreamer, existingIframe;
+
+window.addEventListener("message", (event) => {
+  if (event.data.eventName == "UPDATE_STATE" && event.data.params.quality && !document.hidden)
+    if (/^((?:160|360|480|720|900|960|1080)p(?:30|48|50|60)|chunked)$/.test(event.data.params.quality))
+      localStorage.setItem("embAdbQuality", event.data.params.quality);
+});
+
+const untouchedPages = [(link) => link.endsWith("/directory"), (link) => link.includes("/clip/")];
+
+var observer = new MutationObserver(function (mutations, observer) {
+  var container = document.querySelector(".video-player .tw-absolute");

```

Comment:
Maybe we can create constants for these in case twitch ever decides to change these to fuck with us. For example, it isn't that hard to append a key/randomly generated ID to classes if you're using a build system like webpack which they likely are doing with their front end code.